### PR TITLE
#228 - skip invalid packages on incremental update

### PR DESCRIPTION
**Pull request by olenagerasimova**
For #228 made incremental update skip invalid rpm packages.

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#229](https://github.com/artipie/rpm-adapter/issues/229) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/6aa56373-8f90-4b1d-af11-43cc2865c2e2); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 1]
```diff
@@ -44,58 +49,94 @@
...
-        storage.save(
-            new Key.From("broken.rpm"),
-            new Content.From(
-                broken
-            )
+        storage.save(new Key.From("broken.rpm"), new Content.From(broken)).get();
+        RpmTest.addRpm(storage, "new.rpm", RpmTest.LIBDEFLT);
+        repo.batchUpdate(Key.ROOT).blockingAwait();
+        MatcherAssert.assertThat(
+            countData(tmp),
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/RpmTest.java:74]: @olenagerasimova I would be better if you assert the exact expected metadata file, not just a fragment of it.

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 2]
```diff
@@ -44,58 +49,94 @@
...
-            countData(
-                new String(
-                    new Concatenation(
-                        storage.value(new Key.From(RpmTest.REPOMD)).get()
-                    ).single().blockingGet().array(),
-                    Charset.defaultCharset()
-                )
-            ),
-            new IsEqual<>(countData(repomd))
+            countData(tmp),
```
Comment (paulodamaso) [inline src/test/java/com/artipie/rpm/RpmTest.java:91]: @olenagerasimova Please take a look at the comment above

**Comment by olenagerasimova** [comment]
@paulodamaso thank you for the review, I've created #241 as we need an instrument to validate metadata, assertions will be fixed here after #241 is implemented.

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0crat** [comment]
Code review was too long (10 days), architects (@g4s8, @olenagerasimova[/z](https://www.0crat.com/u/olenagerasimova)) were penalized, see [§55](http://www.zerocracy.com/policy.html#55)

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@paulodamaso please make sure you will find at least 3 issues during next CR, thanks

**Comment by sereshqua** [comment]
@paulodamaso ping

**Comment by paulodamaso** [comment]
@sereshqua I'll try to find at least 3 issues in my next revisions, thank you

**Comment by sereshqua** [comment]
@0crat quality acceptable

**Pull request merged**